apps/election-manager/src/screens/SmartcardsScreen.tsx

```diff
@@ -1,66 +1,101 @@
 import { strict as assert } from 'assert'
 import React, { useContext, useState } from 'react'
 
-import { EventTargetFunction } from '../config/types'
+import { PinPad } from '@votingworks/ui'
+import styled from 'styled-components'
 import AppContext from '../contexts/AppContext'
 
 import NavigationScreen from '../components/NavigationScreen'
 import Prose from '../components/Prose'
 import Button from '../components/Button'
 import Modal from '../components/Modal'
 import Loading from '../components/Loading'
+import { SECURITY_PIN_LENGTH } from '../config/globals'
+
+export const PinEntry = styled.div`
+  text-align: center;
+  color: rgba(71, 167, 75, 1);
+  font-family: monospace;
+  font-size: 1.5em;
+  font-weight: 600;
+`
+
+export const PinPadWrapper = styled.div`
+  display: flex;
+  justify-content: center;
+  margin-top: 10px;
+`
 
 const DefinitionScreen = (): JSX.Element => {
   const { electionDefinition } = useContext(AppContext)
   assert(electionDefinition)
   const { electionData, electionHash } = electionDefinition
 
   const [isProgrammingCard, setIsProgrammingCard] = useState(false)
+  const [isPromptingForAdminPIN, setIsPromptingForAdminPIN] = useState(false)
+  const [currentPin, setCurrentPin] = useState('')
 
-  const programCard: EventTargetFunction = async (event) => {
-    const target = event.currentTarget as HTMLButtonElement
-    const { id } = target.dataset
+  const overrideWriteProtection = async () => {
     setIsProgrammingCard(true)
+    await fetch('/card/write_protect_override', {
+      method: 'post',
+    })
+    window.setTimeout(() => {
```

Comment:
This can be replaced with `await sleep(1000)`, and really should be `await makeCancelable(sleep(1000))`.